packages/plugin-figma-blade-coverage/src/main.ts

```diff
@@ -0,0 +1,429 @@
+    figma.closePlugin();
+  }
+
+  return node;
+};
+
+const renderCoverageCard = async ({
+  mainFrameNode,
+  bladeComponents,
+  nonBladeComponents,
+  nonBladeColorStyles,
+  nonBladeTextStyles,
+  totalLayers,
+}: {
+  mainFrameNode: SceneNode;
+} & CoverageMetrics): Promise<void> => {
+  // these are from payment light theme but it should work as far as the plugin is being run from Razorpay org
+  const COVERAGE_CARD_COMPONENT_KEY = 'c5744871a8db11b02c65b4843d68779f2ff99ed3';
+  const BLADE_INTENT_COLOR_KEYS = {
+    positive: {
+      id: '',
+      key: 'c61aca5db3a21aead10da4889ad2b31c74d93529',
+    },
+    negative: {
+      id: '',
+      key: 'cccac5aac53e828b3be3e8617e462f8ee1a058dd',
+    },
+    notice: {
+      id: '',
+      key: '707d5fdfc748a5fc4777d212ee247bc40a86fe85',
+    },
+  };
+
+  try {
+    const coverageCardComponent = await figma.importComponentByKeyAsync(
+      COVERAGE_CARD_COMPONENT_KEY,
+    );
+    const coverageCardInstance = coverageCardComponent.createInstance();
+    coverageCardInstance.visible = false;
+    coverageCardInstance.x = mainFrameNode.x + 150; // 150 because we want to prevent conflict with the frame name
+    coverageCardInstance.y = mainFrameNode.y - coverageCardComponent.height;
+
+    // import styles for popsitive, negative and notice colors and set their id in BLADE_INTENT_COLOR_KEYS
+    for await (const [intent, intentObject] of Object.entries(BLADE_INTENT_COLOR_KEYS)) {
+      const colorStyle = await figma.importStyleByKeyAsync(intentObject.key);
+      BLADE_INTENT_COLOR_KEYS[intent as 'positive' | 'negative' | 'notice'].id = colorStyle.id;
+    }
+
+    const bladeCoverage = Number((bladeComponents / totalLayers) * 100);
+    let coverageColorIntent = BLADE_INTENT_COLOR_KEYS.negative.id;
+    let bladeCoverageType = 'Very Low 😭';
+    const PROGRESS_BAR_MAX_WIDTH = 254;
+    const bladeCoverageProgress = (bladeCoverage / 100) * PROGRESS_BAR_MAX_WIDTH;
+
+    // calculate coverage type and intent colors for coverage
+    if (bladeCoverage > 70) {
+      bladeCoverageType = 'Good 😊';
+      coverageColorIntent = BLADE_INTENT_COLOR_KEYS.positive.id;
+    } else if (bladeCoverage > 50 && bladeCoverage < 70) {
+      bladeCoverageType = 'Low 😥';
+      coverageColorIntent = BLADE_INTENT_COLOR_KEYS.notice.id;
+    }
+
+    coverageCardInstance.setProperties({
+      'bladeCoverageType#45789:0': bladeCoverageType,
+      'bladeCoverage#45789:1': `${bladeCoverage.toFixed(2)}%`,
+      'totalLayers#45789:2': totalLayers.toString().padStart(2, '0'),
+      'bladeComponents#45789:3': bladeComponents.toString().padStart(2, '0'),
+      'nonBladeComponents#45789:4': nonBladeComponents.toString().padStart(2, '0'),
+      'nonBladeTextStyles#45789:5': nonBladeTextStyles.toString().padStart(2, '0'),
+      'nonBladeColorStyles#45789:6': nonBladeColorStyles.toString().padStart(2, '0'),
+    });
+
+    const detachedCoverageCard = coverageCardInstance.detachInstance();
```

Comment:
🤔 why are we detaching our own Coverage Card?

Reply:
good question but that's because of how figma handles component instances 🤦 

so essentially I want to increase the width of the progress bar and also change the color basis the percentage coverage. now since it's a component I can't change the width based on coverage because you can't change anything in an instance except properties and the properties are just Text properties in this card. 

One way is to add variants for color change but I can't have a variant handle the progress bar width which is dynamic.

Reply:
yes in my research and digging through the docs and forums i couldn't find a better way, and logically as well it's against the instance philosophy. but yeah even I'm not liking it and if you find something better let me know we can replace it.